x-pack/plugin/inference/qa/mixed-cluster/src/javaRestTest/java/org/elasticsearch/xpack/inference/qa/mixed/CohereServiceMixedIT.java

```diff
@@ -0,0 +1,287 @@
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.oneOf;
+
+public class CohereServiceMixedIT extends BaseMixedIT {
+
+    private static final String COHERE_EMBEDDINGS_ADDED = "8.13.0";
+    private static final String COHERE_RERANK_ADDED = "8.14.0";
+    private static final String BYTE_ALIAS_FOR_INT8_ADDED = "8.14.0";
+
+    private static MockWebServer cohereEmbeddingsServer;
+    private static MockWebServer cohereRerankServer;
+
+    @BeforeClass
+    public static void startWebServer() throws IOException {
+        cohereEmbeddingsServer = new MockWebServer();
+        cohereEmbeddingsServer.start();
+
+        cohereRerankServer = new MockWebServer();
+        cohereRerankServer.start();
+    }
+
+    @AfterClass
+    public static void shutdown() {
+        cohereEmbeddingsServer.close();
+        cohereRerankServer.close();
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testCohereEmbeddings() throws IOException {
+        var embeddingsSupported = bwcVersion.onOrAfter(Version.fromString(COHERE_EMBEDDINGS_ADDED));
+        assumeTrue("Cohere embedding service added in " + COHERE_EMBEDDINGS_ADDED, embeddingsSupported);
+
+        final String oldClusterIdInt8 = "old-cluster-embeddings-int8";
+        final String oldClusterIdFloat = "old-cluster-embeddings-float";
+
+        // queue a response as PUT will call the service
+        cohereEmbeddingsServer.enqueue(new MockResponse().setResponseCode(200).setBody(embeddingResponseByte()));
+        put(oldClusterIdInt8, embeddingConfigInt8(getUrl(cohereEmbeddingsServer)), TaskType.TEXT_EMBEDDING);
+        // float model
+        cohereEmbeddingsServer.enqueue(new MockResponse().setResponseCode(200).setBody(embeddingResponseFloat()));
+        put(oldClusterIdFloat, embeddingConfigFloat(getUrl(cohereEmbeddingsServer)), TaskType.TEXT_EMBEDDING);
+
+        var configs = (List<Map<String, Object>>) get(TaskType.TEXT_EMBEDDING, oldClusterIdInt8).get("endpoints");
+        assertEquals("cohere", configs.get(0).get("service"));
+        var serviceSettings = (Map<String, Object>) configs.get(0).get("service_settings");
+        assertThat(serviceSettings, hasEntry("model_id", "embed-english-light-v3.0"));
+        var embeddingType = serviceSettings.get("embedding_type");
+        // An upgraded node will report the embedding type as byte, an old node int8
+        assertThat(embeddingType, Matchers.is(oneOf("int8", "byte")));
+
+        configs = (List<Map<String, Object>>) get(TaskType.TEXT_EMBEDDING, oldClusterIdFloat).get("endpoints");
+        serviceSettings = (Map<String, Object>) configs.get(0).get("service_settings");
+        assertThat(serviceSettings, hasEntry("embedding_type", "float"));
+
+        assertEmbeddingInference(oldClusterIdInt8, CohereEmbeddingType.BYTE);
+        assertEmbeddingInference(oldClusterIdFloat, CohereEmbeddingType.FLOAT);
+
+        {
```

Comment:
These 2 sections appear to be a repeat of the above with a different id. I can see the difference, if it is necessary please add a comment explaining what I am missing. 

Again the names `oldClusterId...` and `upgradedClusterId...` don't apply in the mixed cluster test as there is only the mixed cluster

Reply:
Thanks for catching this!